plugins/out_kinesis_streams/kinesis_api.c

```diff
@@ -97,6 +101,62 @@ static int init_put_payload(struct flb_kinesis *ctx, struct flush *buf,
     return -1;
 }
 
+/*
+ * Count the substring number in string.
+ */
+static int count_substring(char *str1, char *str2){
+    int x = 0;
+    char *p;
+
+    do {
+        p = strstr(str1, str2);
+        if (p != NULL) {
+            str1 = p + 1;
+            x = x + 1;
+        }
+    }
+    while(p != NULL);
+    return x;
+}
+
+/*
+ * Adjust the time_key output format according to time_key_format.
+ */
+static void create_time_format_string(char *out_buff, size_t out_size,
```

Comment:
Should be: char **out_buff, size_t *out_size,

since these are used to get output from the function

Reply:
remember what i said about the out_buf pattern? Look up the `int flb_gzip_compress(void *in_data, size_t in_len,
                      void **out_data, size_t *out_len)` function as an example of this in existing code